sysl2/sysl/ints_view.go

```diff
@@ -0,0 +1,587 @@
+	fmt.Fprintln(v.stringBuilder, "  BorderColor Black")
+	fmt.Fprintln(v.stringBuilder, "  ArrowColor Crimson")
+	if viewParams.highLightColor != "" {
+		fmt.Fprintln(v.stringBuilder, "  BackgroundColor<<highlight>> "+viewParams.highLightColor)
+	}
+	if viewParams.arrowColor != "" {
+		fmt.Fprintln(v.stringBuilder, "  ArrowColor "+viewParams.arrowColor)
+	}
+
+	if viewParams.indirectArrowColor != "" && viewParams.indirectArrowColor != "none" {
+		fmt.Fprintln(v.stringBuilder, "  ArrowColor<<indirect>> "+viewParams.indirectArrowColor)
+	}
+	fmt.Fprintln(v.stringBuilder, "}")
+
+	nameMap := map[string]string{}
+	if args.clustered || viewParams.endptAttrs["view"].GetS() == "clustered" {
+		nameMap = v.buildClusterForComponentView(params.apps)
+	}
+	v.drawComponentView(viewParams, params, nameMap)
+	fmt.Fprintln(v.stringBuilder, "@enduml")
+	return v.stringBuilder.String()
+}
+
+func GenerateView(args *Args, params *IntsParam, mod *sysl.Module) string {
+	var stringBuilder strings.Builder
+	v := MakeIntsDiagramVisitor(mod, &stringBuilder, params.highlights, args.project)
+	restrictBy := ""
+	if params.endpt.Attrs["restrict_by"] != nil {
+		restrictBy = params.endpt.Attrs["restrict_by"].GetS()
+	}
+
+	appAttrs := params.app.Attrs
+	endptAttrs := params.endpt.Attrs
+	highLightColor := appAttrs["highlight_color"].GetS()
+	arrowColor := appAttrs["arrow_color"].GetS()
+	indirectArrowColor := appAttrs["indirect_arrow_color"].GetS()
+
+	diagramTitle := ""
+	attrs := map[string]string{
+		"epname":     params.endpt.Name,
+		"eplongname": params.endpt.LongName,
+	}
+	if appAttrs["title"].GetS() != "" {
+		diagramTitle = ParseFmt(attrs, appAttrs["title"].GetS())
+	} else {
+		diagramTitle = ParseFmt(attrs, args.title)
+	}
+
+	viewParams := &viewParams{
+		restrictBy:         restrictBy,
+		endptAttrs:         endptAttrs,
+		highLightColor:     highLightColor,
+		arrowColor:         arrowColor,
+		indirectArrowColor: indirectArrowColor,
+		diagramTitle:       diagramTitle,
+	}
+
+	fmt.Fprintln(&stringBuilder, "''''''''''''''''''''''''''''''''''''''''''")
+	fmt.Fprintln(&stringBuilder, "''                                      ''")
+	fmt.Fprintln(&stringBuilder, "''  AUTOGENERATED CODE -- DO NOT EDIT!  ''")
+	fmt.Fprintln(&stringBuilder, "''                                      ''")
+	fmt.Fprintln(&stringBuilder, "''''''''''''''''''''''''''''''''''''''''''")
+	fmt.Fprintln(&stringBuilder)
+
+	if args.epa || endptAttrs["view"].GetS() == "epa" {
+		return v.generateStateView(args, *viewParams, params)
+	} else {
+		return v.generateComponentView(args, *viewParams, params)
+	}
+}
+
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+
+func getAttrs(m *sysl.Module, appName string) map[string]string {
+	val := map[string]string{}
+	if app, ok := m.Apps[appName]; ok {
+		attrs := app.Attrs
+		for k, v := range attrs {
+			val["@"+k] = v.GetS()
+		}
```

Comment:
@JackLeeHal Dont build another copy of the map. Remove leading at `@`

Reply:
@sahejsingh changed to `for k, v := range app.Attrs {`

Reply:
That map is for appfmt and epfmt's parser, eg. `%(@ACTIVE_id?//«%(@ACTIVE_id)»//)`, that variable needed extra `@`

Reply:
cant you add `@` as prefix while looking up?